pages/voting-guides/index.tsx

```diff
@@ -0,0 +1,149 @@
+  const {user} = guide;
+  const {firstName, lastName, avatarUrl, username} = user;
+  const name = firstName ? `${firstName} ${!!lastName ? lastName:''}`: username;
+  return(
+  <div className={`${styles.roundedCard} ${styles.votingCardContainer}`}>
+  <div className={styles.guideContainer}>
+    <div className={styles.avatarContainer}>
+      <Avatar src={avatarUrl || ""} size={80} fallbackSrc="https://www.gravatar.com/avatar/" alt={name} />
+      <h4>{name}</h4>
+    </div>
+    <div className={styles.buttonWrapper}>
+      <Button large primary theme="blue" label="Edit">Edit</Button>
+      <span className={styles.shareButton}>
+        <Button large primary theme="yellow" label="Share">Share</Button>
+      </span>
+    </div>
+  </div>
+</div>);
+};
+
+interface VotingGuideElection {
+  date: string,
+  description?: string,
+  guides: VotingGuideResult[],  
+  id: Scalars['ID'],
+  title: string,
+}
+
+const getElectionsFromGuides = (guides: VotingGuideResult[]): VotingGuideElection[]=>{
+  if (guides){
+    return guides.reduce<VotingGuideElection[]>((result, value) => {
+      const i = result.findIndex(v=>v.id===value.electionId);
+      if (i<0) {
+        const electionInfo = {
+          id: value.electionId,
+          title: value.election.title,
+          description: value.election.description,
+          date: value.election.electionDate,
+          guides: [value]
+        } as VotingGuideElection;
+        result.push(electionInfo);
+      } else {
+        result[i]?.guides.push(value);
+      }
+      return result;
+    }, [] as VotingGuideElection[]);
+  } else return [] as VotingGuideElection[];
+};
+
+const VotingGuides: NextPage<{ mobileNavTitle?: string }> = ({mobileNavTitle}) => {
+
+  const user = useAuth({redirectTo:"/login"}).user; //user will not be null under this line, due to the conditional redirect
+  const votingGuidesByUserId = useVotingGuidesByUserIdQuery({userId:user?.id || ""}); 
+  const [elections, setElections] = useState<VotingGuideElection[]>([]);
+
+  useEffect(()=>{
+    const {data, isError, error} = votingGuidesByUserId;
+  if (isError){
+    alert("Error getting Voting Guides");
+    console.error("Error getting Voting Guides", error);
+  }
+  else{
+    if (data && data.votingGuidesByUserId.length) {
+      const newVotingGuides = data.votingGuidesByUserId as VotingGuideResult[];
+      const elections = getElectionsFromGuides(newVotingGuides);
+      console.log('yes data', elections);
+      setElections(elections);
+    } else {
+      votingGuidesByUserId.refetch().then(res=>{
+        const elections = getElectionsFromGuides(res.data?.votingGuidesByUserId as VotingGuideResult[]);
+        console.log('no data', elections);
+        setElections(elections);
+      });
+    }
+  };
+  },[]);
+
+  const upsertVotingGuide = useUpsertVotingGuideMutation();
+
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
```

Comment:
lets use `react-hook-form` for this instead of manually managing this state

Reply:
Would rather not bother since this whole section is entirely temporary and will be removed as soon as we can add voting guides elsewhere